Adding cuDNN example

lindazy opened the pull request:
Example to run theano with gpu.

lindazy committed:
Adding cuDNN example

frenchwr commented:
Thanks for the contribution, @lindazy. A few requests before we pull this into the ACCRE Python repo:
1. Could you include the `check.py` file in the repo so users can test this job?
2. It seems like `theano` would be a more appropriate name for the directory rather than `cuDNN`.
3. Could you update the example so that it does not provide instructions for running on a dedicated box? In general (your group is currently the only exception) users should only be submitting GPU jobs through SLURM. You might even include the steps outlining the virtualenv activation and theano installation inside of the SLURM script, and mention that this can be commented out if theano is already installed.

Note that you do not need to submit a new pull requests when incorporating these changes. When you push the changes to lindazy:master the changes will automatically be reflected in this pull request.

lindazy commented:
I'll do that tonight, thanks!

---

From: Will French [notifications@github.com]
Sent: Tuesday, July 12, 2016 3:52 PM
To: accre/Python
Cc: Zhang, Linda; Mention
Subject: Re: [accre/Python] Adding cuDNN example (#6)

Thanks for the contribution, @lindazyhttps://github.com/lindazy. A few requests before we pull this into the ACCRE Python repo:
1.  Could you include the check.py file in the repo so users can test this job?
2.  It seems like theano would be a more appropriate name for the directory rather than cuDNN.
3.  Could you update the example so that it does not provide instructions for running on a dedicated box? In general (your group is currently the only exception) users should only be submitting GPU jobs through SLURM. You might even include the steps outlining the virtualenv activation and theano installation inside of the SLURM script, and mention that this can be commented out if theano is already installed.

Note that you do not need to submit a new pull requests when incorporating these changes. When you push the changes to lindazy:master the changes will automatically be reflected in this pull request.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/accre/Python/pull/6#issuecomment-232177424, or mute the threadhttps://github.com/notifications/unsubscribe/AIkwk-CAxNOdOkKzBSymynrsg7AVPDPoks5qU_6JgaJpZM4JKyNo.

lindazy committed:
Added theano example

lindazy commented:
I've made the changes I think. Let me know if there's everything seems okay!

---

From: Zhang, Linda
Sent: Tuesday, July 12, 2016 5:06 PM
To: accre/Python; accre/Python
Cc: Mention
Subject: RE: [accre/Python] Adding cuDNN example (#6)

I'll do that tonight, thanks!

---

From: Will French [notifications@github.com]
Sent: Tuesday, July 12, 2016 3:52 PM
To: accre/Python
Cc: Zhang, Linda; Mention
Subject: Re: [accre/Python] Adding cuDNN example (#6)

Thanks for the contribution, @lindazyhttps://github.com/lindazy. A few requests before we pull this into the ACCRE Python repo:
1.  Could you include the check.py file in the repo so users can test this job?
2.  It seems like theano would be a more appropriate name for the directory rather than cuDNN.
3.  Could you update the example so that it does not provide instructions for running on a dedicated box? In general (your group is currently the only exception) users should only be submitting GPU jobs through SLURM. You might even include the steps outlining the virtualenv activation and theano installation inside of the SLURM script, and mention that this can be commented out if theano is already installed.

Note that you do not need to submit a new pull requests when incorporating these changes. When you push the changes to lindazy:master the changes will automatically be reflected in this pull request.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/accre/Python/pull/6#issuecomment-232177424, or mute the threadhttps://github.com/notifications/unsubscribe/AIkwk-CAxNOdOkKzBSymynrsg7AVPDPoks5qU_6JgaJpZM4JKyNo.

frenchwr commented:
Looking good! I was able to successfully run this example after making a few changes. I have a few more suggestions/questions just to make the example as widely applicable to ACCRE users:
- You can remove the cuDNN subdirectory from the repo.
- I would add this step to the README.md:

```
cp example_theanorc.txt ~/.theanorc
```
- Instead of adding the example_bashrc.txt file I would just add that in the SLURM script. This often is more convenient if you are running different types of jobs where each job needs a slightly different environment. Here's what my SLURM script looks like:

```
#!/bin/bash
#SBATCH --account=accre_gpu
#SBATCH --partition=maxwell
#SBATCH --gres=gpu:1
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --mem=10G
#SBATCH --time=2:00:00

setpkgs -a python3.4.2 # includes theano
setpkgs -a cudnn7.0-v3
setpkgs -a cuda7.0

python check.py
```
- I noticed you aded OpenBLAS to your LD_LIBRARY_PATH. Is OpenBLAS required for certain features within theano to function correctly?

lindazy committed:
Added theano example

lindazy commented:
I actually believe you do not need BLAS. I've removed it, and updated the repo with the changes (forgot to commit the remove cuDNN subdirectory before oops). How's it look now?

---

From: Will French [notifications@github.com]
Sent: Wednesday, July 13, 2016 4:27 PM
To: accre/Python
Cc: Zhang, Linda; Mention
Subject: Re: [accre/Python] Adding cuDNN example (#6)

Looking good! I was able to successfully run this example after making a few changes. I have a few more suggestions/questions just to make the example as widely applicable to ACCRE users:
-   You can remove the cuDNN subdirectory from the repo.
-   I would add this step to the README.md:

cp example_theanorc.txt ~/.theanorc
-   Instead of adding the example_bashrc.txt file I would just add that in the SLURM script. This often is more convenient if you are running different types of jobs where each job needs a slightly different environment. Here's what my SLURM script looks like:

#!/bin/bash
#SBATCH --account=accre_gpu
#SBATCH --partition=maxwell
#SBATCH --gres=gpu:1
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --mem=10G
#SBATCH --time=2:00:00

setpkgs -a python3.4.2 # includes theano
setpkgs -a cudnn7.0-v3
setpkgs -a cuda7.0

python check.py
-   I noticed you aded OpenBLAS to your LD_LIBRARY_PATH. Is OpenBLAS required for certain features within theano to function correctly?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHubhttps://github.com/accre/Python/pull/6#issuecomment-232492099, or mute the threadhttps://github.com/notifications/unsubscribe/AIkwk1SnjJl_13kL-C44fa6lDZNVKs6Bks5qVVhJgaJpZM4JKyNo.

frenchwr commented:
Looks great! Thanks again.
